joint_limits_controller/src/joint_limiter.cpp

```diff
@@ -0,0 +1,85 @@
+#include <joint_limits_controller/limited_joint_handle.h>
+
+double JointLimiter::limitBounds(double value, double min, double max){
+    if(value > max) return max;
+    else if (value < min) return min;
+    return value;
+}
+
+std::pair<double,double> JointLimiter::getSoftBounds(double value, double k, double lower, double upper){
+    return std::make_pair(-k*(value-lower), -k*(value-upper));
+}
+
+void PositionJointLimiter::enforceLimits(const double& period, const Limits &limits, const double pos, const double vel, const double eff, double &cmd) {
+    double last_command = 0;
+
+    if(!last_command_.get(last_command)) last_command = pos; // fallback to actual position
+
+    if(limits.hasSoftLimits()){
+        std::pair<double, double> soft_bounds = limits.getVelocitySoftBounds(last_command);
+
+        // try to limit velocity to hard bounds
```

Comment:
The check for hard limits is superfluous here, since it already done in `getVelocitySoftBounds`